components/list/list-item.js

```diff
@@ -342,10 +344,34 @@ class ListItem extends RtlMixin(LitElement) {
 		}));
 	}
 
+	_handleBlurLink() {
+		this._focusingLink = false;
+	}
+
 	_handleCheckboxChange(e) {
 		this.setSelected(e.target.checked);
 	}
 
+	_handleFocusIn() {
+		this._focusing = true;
+	}
+
+	_handleFocusLink() {
+		this._focusingLink = true;
+	}
```

Comment:
This is needed separate from capturing focus on the list-item. This is specific to the anchor.